core/src/main/java/com/oath/oak/OakRKeyReferenceImpl.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2018 Oath Inc.
+ * Licensed under the terms of the Apache 2.0 license.
+ * Please see LICENSE file in the project root for terms.
+ */
+
+package com.oath.oak;
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.util.function.Function;
+
+public class OakRKeyReferenceImpl implements OakRBuffer {
+
+    private int blockID = 0;
+    private int keyPosition = 0;
+    private int keyLength = 0;
+    private final MemoryManager memoryManager;
+
+    private ByteBuffer bb = null;
+
+    // The OakRKeyReferBufferImpl user accesses OakRKeyReferBufferImpl
+    // as it would be a ByteBuffer with initially zero position.
+    // We translate it to the relevant ByteBuffer position, by adding keyPosition to any given index
+
+    OakRKeyReferenceImpl(MemoryManager memoryManager) {
+        this.memoryManager = memoryManager;
+    }
+
+    public void setBlockID(int blockID) {
```

Comment:
Do you need to nullify bb if blockid is not changed?

If these 3 methods are always called together `setBlockID` `setKeyPosition` and `setKeyLength` then maybe have a single method with 3 parameters and invalidate bb just once.

opening an option to set only some of them is a potential for bugs

Reply:
You also need to add the disclaimer that this code is for use only in special cases like flush where only a single reader go over the entries and it is not updated during the traversal.

Reply:
fixed